src/libraries/System.Private.CoreLib/src/System/Globalization/CultureData.cs

```diff
@@ -1393,7 +1394,7 @@ internal string[] LongTimes
                 {
                     Debug.Assert(!GlobalizationMode.Invariant);
 
-                    string[]? longTimes = GlobalizationMode.UseNls ? NlsGetTimeFormats() : IcuGetTimeFormats();
+                    string[]? longTimes = GetTimeFormatsCore();
```

Comment:
>GetTimeFormatsCore [](start = 42, length = 18)

I think the user can override the default time format and not all time formats. in other word, we need to read the whole list from ICU and the default one from NLS. this will apply for all properties returning a list.

Reply:
Ok, makes sense.

Reply:
Will an running application see new time format if user changes the time format on the fly?

Reply:
It will have the same behavior as it has now, currently we already cache the values so that wouldn’t be a different behavior.

I think applications need to call CultureInfo.ClearCache().

Reply:
I would expect that PowerShell (e.g. Get-ChildItem's console output, currently affected by this issue) takes user changes into account on the fly.

Reply:
@safern Do you say about ICU only behavior? Was there a cache for NLS too (in .Net 3)?

Reply:
There is no change in the behavior than before here. The default user locale was cached before and can be reset by calling ClearCachedData. 
@sba923 @iSazonov I would suggest after @safern merge this change, you give it a try and let's know if you see any regression. We'll be happy to address it.